Spec/Tests/RealtimeClientConnectionTests.swift

```diff
@@ -2234,41 +2234,40 @@ class RealtimeClientConnectionTests: XCTestCase {
             client.close()
         }
 
-        var totalRetry = 0
+        var retryCount = 0
         waitUntil(timeout: testTimeout) { done in
-            let partialDone = AblyTests.splitDone(2, done: done)
-            var start: NSDate?
-
-            client.connection.once(.disconnected) { stateChange in
+            var disconnectedAt: Date!
+            
+            client.connection.on(.disconnected) { stateChange in
                 expect(stateChange.reason!.message).to(contain("timed out"))
                 expect(stateChange.previous).to(equal(ARTRealtimeConnectionState.connecting))
-                expect(stateChange.retryIn).to(beCloseTo(options.disconnectedRetryTimeout))
-                partialDone()
-                start = NSDate()
+                
+                let expectedDelay = ARTRealtimeInternal.retryDelay(fromTimeout: options.disconnectedRetryTimeout, retryCount: retryCount + 1)
```

Comment:
I think that if we're going to use our own code inside the tests, then we probably need to have some unit tests for that code separately too.

Reply:
https://github.com/ably/ably-cocoa/pull/1520/commits/5ce76ddd9b961c657e411f91c4beb26f5b10e386